go.mod

```diff
@@ -0,0 +1,65 @@
+module github.com/libp2p/go-libp2p-kad-dht
```

Comment:
This package has tags greater than v2.*.*, therefore the migration path to go mod is as follows:
1. Branch master with the name `vN`, `N` being whatever the latest major version is, i.e. `v4` in this case.
2. Make sure that the latest git tag is level with the latest gx version, if it isn't add it, if the latest commit is unversioned in gx, patch increment the git tag.
3. Create branch for this PR, 
  3.1. `go mod init`
  3.2. Edit the `go.mod` file to have the appropriate version, i.e. `/v4`, appended to the end of the `module` directive, i.e. `module github.com/libp2p/go-libp2p-kad-dht/v4`
  3.3. Replace all import statements inside the package with the new versioned import statement, i.e. 
`sed -i 's:"github.com/libp2p/go-libp2p-kad-dht:"github.com/libp2p/go-libp2p-kad-dht/v4:g' **/*.go`
`sed -i 's:"github.com/libp2p/go-libp2p-kad-dht:"github.com/libp2p/go-libp2p-kad-dht/v4:g' *.go`
  3.4. Run `go get ./...`, this will populate the `go.mod` file with deps and also surface any issues with imports of the new versioned import path of this package.
  3.5 Run `go test all` to make sure that the chosen deps are suitable.
4. Remove `package.json` and `.gx` 
5. Update any CI system files
6. Update `Makefile` to use `env GO111MODULE=on go get ./...` for `deps` target